test-frame-common/src/main/java/io/skodjob/testframe/clients/KubeClient.java

```diff
@@ -98,6 +105,185 @@ public List<HasMetadata> readResourcesFromFile(InputStream is) throws IOExceptio
         }
     }
 
+    /**
+     * Check if namespace exists in current cluster
+     * @param namespace namespace name
+     * @return true if namespace exists
+     */
+    public boolean namespaceExists(String namespace) {
+        return client.namespaces().list().getItems().stream().map(n -> n.getMetadata().getName())
+                .toList().contains(namespace);
+    }
+
+    /**
+     * Create resources from file and apply modifier
+     * @param namespace dest namespace
+     * @param is stream of data
+     * @param modifier modifier method
+     * @throws IOException exception
+     */
+    public void create(String namespace, InputStream is, Function<HasMetadata, HasMetadata> modifier)
```

Comment:
`create` name is a bit misleading, as you are updating the resource here as well (in case that the resource already exists).
Also, is a good practice to have these methods separated, because when you are doing some operation in the test (for example creating a CR), you know that it's there and it should be just updated.